### PR TITLE
[1.19.x] Update and Regenerate Datapacks

### PR DESCRIPTION
**Pull request by ChampionAsh5357**
In response to [Lex's comment](https://github.com/MinecraftForge/MinecraftForge/pull/9414#issuecomment-1483359534), updates the `main` and `test` source sets to generate their `pack.mcmeta` using `PackMetadataGenerator`. In addition, this regenerates the JSONs to push any changes that would be necessary.

This does not cover the example mod as there isn't really a good solution for that without creating a separate source set to run the generation or parsing from the `main` or `test` source set on push. If you have any good solutions, I'm open to making a quick implementation for it.

### CODE REVIEW

File: src/test/resources/META-INF/mods.toml [hunk 1]
```diff
@@ -175,8 +175,6 @@ modId="remove_tag_datagen_test"
 [[mods]]
 modId="tag_based_tool_types"
 [[mods]]
-modId="check_spawn_event_test"
-[[mods]]
```
Comment (sciwhiz12) [inline src/test/resources/META-INF/mods.toml:179]: Noting for others: the test mod class for this was removed by #9133, which forgot to remove this entry.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
I cloned your PR and locally ran both data generation tasks (main and test), and it seems that the test sourceset's data generation task gives me one file not accounted for in this PR: `src/generated_test/resources/assets/data_gen_test/models/item/test_block_model.json`. Could you please look into that?

**Comment by sciwhiz12** [comment]
Regarding the MDK's pack metadata, perhaps what could be done is having the test sourceset generate a pack metadata under a folder (using a cutom `PackOutput`), which is then copied over to the MDK when the ZIP file is created. We could also use the main sourceset for this, and configuring the sourceset to exclude that specific folder if possible so it is not copied to the main Forge JAR.

**Comment by ChampionAsh5357** [comment]
> I cloned your PR and locally ran both data generation tasks (main and test), and it seems that the test sourceset's data generation task gives me one file not accounted for in this PR: `src/generated_test/resources/assets/data_gen_test/models/item/test_block_model.json`. Could you please look into that?

It seems to be an issue with files not generating when the completable future isn't activated, so I've fixed that now.


> Regarding the MDK's pack metadata, perhaps what could be done is having the test sourceset generate a pack metadata under a folder (using a cutom `PackOutput`), which is then copied over to the MDK when the ZIP file is created. We could also use the main sourceset for this, and configuring the sourceset to exclude that specific folder if possible so it is not copied to the main Forge JAR.

So, I ended up creating a separate task for this which generates stuff for the mdk in the test set. I chose test as the datagen is not really applicable to main Forge. Additionally, it's a separate task as otherwise the datagen would fail due to the providers having the same name, which can not be changed due to the finality of the method. After that, I just copy the files into the mdk such that they are in the correct location.

File: src/test/java/net/minecraftforge/mdk/MdkDatagen.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.mdk;
+
```
Comment (sciwhiz12) [inline src/test/java/net/minecraftforge/mdk/MdkDatagen.java:7]: Looks like an extra blank line was accidentally inserted here.

File: src/test/resources/META-INF/mods.toml [hunk 3]
```diff
@@ -285,5 +286,4 @@ modId="trade_with_villager_event_test"
 [[mods]]
 modId="chunk_event_load_new_chunk_test"
 
-
```
Comment (sciwhiz12) [inline src/test/resources/META-INF/mods.toml]: Please re-add this blank line.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Looks and works fine. Just have two minor whitespace changes.

**Comment by ChampionAsh5357** [comment]
Done, should be nothing else here.

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**